### PR TITLE
Implement Index.nunique

### PR DESCRIPTION
**Pull request by RainFung**
Implement Index.nunique by F.countDistinct

### CODE REVIEW

File: databricks/koalas/missing/indexes.py [hunk 1]
```diff
@@ -73,7 +73,6 @@ class _MissingPandasLikeIndex(object):
     is_type_compatible = unsupported_function('is_type_compatible')
     join = unsupported_function('join')
     map = unsupported_function('map')
-    nunique = unsupported_function('nunique')
```
Comment (itholic) [inline databricks/koalas/missing/indexes.py]: could you add `MultiIndex.nunique` also and just raise `NotImplementedError` for `MultiIndex` like pandas?

```python
>>> pidx = pd.MultiIndex.from_tuples([('a', 1), ('b', 2)])
>>> pidx.nunique()
Traceback (most recent call last):
...
NotImplementedError: isna is not defined for MultiIndex
```

File: databricks/koalas/tests/test_indexes.py [hunk 2]
```diff
@@ -357,3 +357,10 @@ def test_multiindex_isna(self):
                 NotImplementedError,
                 "notna is not defined for MultiIndex"):
             kidx.notnull()
+
+    def test_index_nunique(self):
+        pidx = pd.DataFrame({'a': ['a', 'b', 'c', 'd']}, index=[1, 1, 2, None]).index
+        kidx = ks.DataFrame({'a': ['a', 'b', 'c', 'd']}, index=[1, 1, 2, None]).index
```
Comment (itholic) [inline databricks/koalas/tests/test_indexes.py]: nit: you can just like below.

```python
pidx = pd.Index([1, 1, 2, None])
kidx = ks.Index([1, 1, 2, None])
```

File: databricks/koalas/tests/test_indexes.py [hunk 3]
```diff
@@ -357,3 +357,17 @@ def test_multiindex_isna(self):
...
+
+        self.assert_eq(pidx.nunique(), kidx.nunique())
+        self.assert_eq(pidx.nunique(dropna=True), kidx.nunique(dropna=True))
+
+    def test_multiindex_nunique(self):
+        kidx = ks.MultiIndex.from_tuples([('a', 'x', 1), ('b', 'y', 2), ('c', 'z', 3)])
+        with self.assertRaisesRegex(
+                NotImplementedError,
+                "notna is not defined for MultiIndex"):
+            kidx.notnull()
```
Comment (itholic) [inline databricks/koalas/tests/test_indexes.py]: maybe

```python
        with self.assertRaisesRegex(
                NotImplementedError,
                "isna is not defined for MultiIndex"):
            kidx.nunique()
```

?? :)

File: databricks/koalas/indexes.py [hunk 4]
```diff
@@ -588,6 +588,48 @@ def dropna(self):
         kdf = DataFrame(internal)
         return Index(kdf) if type(self) == Index else MultiIndex(kdf)
 
+    def nunique(self, dropna=True):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: @RainFung sorry for my late response. There's `Series.nunique` implementation. Can we move that implementation to `IndexOpsMixin` to support this in Index?

File: docs/source/reference/indexing.rst [hunk 5]
```diff
@@ -53,6 +53,7 @@ Modifying and computations
    Index.max
    Index.rename
    Index.unique
+   Index.nunique
```
Comment (HyukjinKwon) [inline docs/source/reference/indexing.rst]: Should be good to add it in `MultiIndex` as well.

Comment (RainFung) [reply docs/source/reference/indexing.rst]: MultiIndex.nunique has been deprecated in `pandas 0.26 API`
https://dev.pandas.io/docs/reference/indexing.html

File: databricks/koalas/indexes.py (hunk 4 above)
Comment (RainFung) [reply databricks/koalas/indexes.py]: OK, i will implement it based on IndexOpsMixin

File: docs/source/reference/indexing.rst (hunk 5 above)
Comment (HyukjinKwon) [reply docs/source/reference/indexing.rst]: Ah, sure. sounds good!

**Comment by HyukjinKwon** [comment]
@RainFung can you resolve conflicts?

**Comment by HyukjinKwon** [comment]
@RainFung can you resolve conflicts?  Looks good.

**Pull request merged**